packages/block-library/src/legacy-widget/edit/form.js

```diff
@@ -1,143 +1,73 @@
-	const setFormDataDebounced = useCallback( debounce( setFormData, 300 ), [
-		setFormData,
-	] );
+export default function Form( {
+	title,
+	isVisible,
+	id,
+	idBase,
+	instance,
+	onChangeInstance,
+	onChangeHasPreview,
+} ) {
+	const ref = useRef();
+	const outgoingInstances = useRef( new Set() );
 
-	return (
-		<Control
-			id={ id }
-			idBase={ idBase }
-			content={ content }
-			onChange={ setFormDataDebounced }
-			onSave={ setFormData }
-			// Force a remount when the widget's form HTML changes. This clears
-			// out any mutations to the DOM that widget scripts have made.
-			key={ content }
-		/>
-	);
-}
+	const { createNotice } = useDispatch( noticesStore );
 
-function Control( { id, idBase, content, onChange, onSave } ) {
-	const controlRef = useRef();
-	const formRef = useRef();
-
-	// Trigger 'widget-added' when widget is ready. This event is what widgets'
-	// scripts use to initialize, attach events, etc. The event must be fired
-	// using jQuery's event bus as this is what widget scripts expect. If jQuery
-	// is not loaded, do nothing - some widgets will still work regardless.
 	useEffect( () => {
-		if ( ! window.jQuery ) {
+		if ( outgoingInstances.current.has( instance ) ) {
+			outgoingInstances.current.delete( instance );
 			return;
 		}
 
-		const { jQuery: $ } = window;
+		const control = createControl( {
+			id,
+			idBase,
+			instance,
+			onChangeInstance( nextInstance ) {
+				outgoingInstances.current.add( nextInstance );
+				onChangeInstance( nextInstance );
+			},
+			onChangeHasPreview,
+			onError( error ) {
+				createNotice(
+					'error',
+					error?.message ??
+						__(
+							'An error occured while fetching or updating the widget.'
+						)
+				);
+			},
+		} );
 
-		if ( content ) {
-			$( document ).trigger( 'widget-added', [
-				$( controlRef.current ),
-			] );
-		}
-	}, [ content ] );
+		ref.current.appendChild( control.element );
 
-	// Prefer jQuery 'change' event instead of the native 'change' event because
-	// many widgets use jQuery's event bus to trigger an update.
-	useEffect( () => {
-		const handler = () => onChange( serializeForm( formRef.current ) );
-
-		if ( window.jQuery ) {
-			const { jQuery: $ } = window;
-			$( formRef.current ).on( 'change', null, handler );
-			return () => $( formRef.current ).off( 'change', null, handler );
-		}
-
-		formRef.current.addEventListener( 'change', handler );
-		return () => formRef.current.removeEventListener( 'change', handler );
-	}, [ onChange ] );
-
-	// Non-multi widgets can be saved via a Save button.
-	const handleSubmit = ( event ) => {
-		event.preventDefault();
-		onSave( serializeForm( event.target ) );
-	};
-
-	// We can't use the real widget number as this is calculated by the server
-	// and we may not ever *actually* save this widget. Instead, use a fake but
-	// unique number.
-	const number = useInstanceId( Control );
+		return () => {
+			ref.current.removeChild( control.element );
+			control.destroy();
```

Comment:
Nit: We could embed `element.remove()` in `control.destroy()` and just call `destroy` here.